tesseract_common/include/tesseract_common/plugin_loader.hpp

```diff
@@ -110,6 +114,12 @@ std::shared_ptr<PluginBase> PluginLoader::instantiate(const std::string& plugin_
     }
   }
 
+  for (const auto& library : plugins_local)
```

Comment:
I believe you found a bug, The above for loop should use `plugins_local` instead of search_libraries. I believe this can be removed if that change is made.

Reply:
It looks like there is another for loop above that should also be using the same variable.

Reply:
Yeah this function has a couple strange issues that need to be cleaned up